packages/web-components/src/components/ic-data-table/ic-data-table.tsx

```diff
@@ -0,0 +1,498 @@
+    sortOrders: IcDataTableSortOrderOptions[];
+    defaultColumn?: string;
+  } = {
+    sortOrders: ["unsorted", "ascending", "descending"],
+    defaultColumn: "",
+  };
+
+  /**
+   * If `true`, column headers will remain at the top of the table when scrolling vertically.
+   */
+  @Prop() stickyColumnHeaders?: boolean = false;
+
+  /**
+   * If `true`, row headers will remain to the left when scrolling horizontally.
+   */
+  @Prop() stickyRowHeaders?: boolean = false;
+
+  componentWillLoad(): void {
+    this.rowsPerPage = Number(this.paginationOptions.itemsPerPage[0].value);
+    this.previousRowsPerPage = this.rowsPerPage;
+    this.toRow = this.rowsPerPage;
+    this.sortedColumn = this.sortOptions.defaultColumn;
+    this.sortedColumnOrder = this.sortOptions.sortOrders[0];
+  }
+
+  componentDidLoad(): void {
+    const tableElement = this.el.shadowRoot.querySelector("table");
+    const tableContainer = this.el.shadowRoot.querySelector(".table-container");
+    if (
+      tableElement?.clientHeight > tableContainer?.clientHeight ||
+      tableElement?.clientWidth > tableContainer?.clientWidth
+    ) {
+      this.scrollable = true;
+    }
+  }
+
+  @Listen("icItemsPerPageChange")
+  handleItemsPerPageChange(ev: CustomEvent): void {
+    this.previousRowsPerPage = this.rowsPerPage;
+    this.rowsPerPage = ev.detail.value;
+  }
+
+  @Listen("icPageChange")
+  handlePageChange(ev: CustomEvent): void {
+    this.fromRow = (ev.detail.value - 1) * this.rowsPerPage;
+    this.toRow = this.fromRow + this.rowsPerPage;
+    const tableRowsContainer = this.el.shadowRoot.querySelector(
+      ".table-row-container"
+    );
+    if (this.previousRowsPerPage === this.rowsPerPage) {
+      tableRowsContainer.scrollTop = 0;
+    } else if (this.previousRowsPerPage < this.rowsPerPage) {
+      tableRowsContainer.scrollTop = this.scrollOffset;
+      this.previousRowsPerPage = this.rowsPerPage;
+    } else {
+      this.previousRowsPerPage = this.rowsPerPage;
+    }
+  }
+
+  @Listen("click", { target: "window" })
+  clickListener(ev: MouseEvent): void {
+    if (ev.target !== this.el) this.selectedRow = undefined;
+  }
+
+  private isObject = (value: any) => typeof value === "object";
+
+  private notDefaultDensity = () => this.density !== "default";
+
+  private createCells = (row: object) => {
+    const rowValues = Object.values(row);
+    const rowKeys = Object.keys(row);
+    let rowAlignment: string;
+    let rowEmphasis: string;
+    const headerIndex = rowKeys.indexOf("header");
+    if (headerIndex > -1) {
+      rowAlignment = this.getObjectValue(
+        rowValues[headerIndex],
+        "rowAlignment"
+      );
+      rowEmphasis = this.getObjectValue(rowValues[headerIndex], "emphasis");
+    }
+    return rowValues.map((cell, index) => {
+      const column = this.columns[index];
+      const cellValue = (key: string) => this.getObjectValue(cell, key);
+
+      const getAlignment = (alignment: "horizontal" | "vertical") => {
```

Comment:
Feel as though this can be outside of the `map` and pass through `cell` in addition to `alignment`. This will also make it possible to unit test

Reply:
Not sure this is necessary. This code is only repeated in this specific map function, and contains values that are only within the scope of the map function. Keeping it here would prevent unnecessarily passing through the same value multiple times.

Reply:
How would you unit test the function?